src/Components/Endpoints/src/Builder/RazorComponentEndpointDataSource.cs

```diff
@@ -77,11 +85,42 @@ private void UpdateEndpoints()
     {
         var endpoints = new List<Endpoint>();
         var context = _builder.Build();
+        if (_renderModes.Count == 0)
+        {
+            _renderModes.AddRange(context.ResolveRenderModes());
```

Comment:
What if you explicitly don't want any modes?

Reply:
Do we need a Static render mode to indicate that we don't want any interactivity (so the app is purely SSR)

Reply:
> Does it make sense if you have a server or webassembly component to not include the endpoints needed for them to work?

Yes, if you only want SSR.

> Do we need a Static render mode to indicate that we don't want any interactivity (so the app is purely SSR)

I would expect this to be expressed by not saying you want Server/WebAssembly.

Reply:
If you are doing SSR, either don't have your components declare a render mode, or set the render mode on the pipeline to static explicitly. We could do nothing instead of throwing if you have not configured anything. However, the consequence of that is that it will fail silently at runtime.